src/diffusers/schedulers/scheduling_euler_ancestral_discrete.py

```diff
@@ -239,6 +249,26 @@ def set_timesteps(self, num_inference_steps: int, device: Union[str, torch.devic
         else:
             self.timesteps = torch.from_numpy(timesteps).to(device=device)
 
+        self._step_index = None
+
+    # Copied from diffusers.schedulers.scheduling_euler_discrete.EulerDiscreteScheduler._init_step_index
+    def _init_step_index(self, timestep):
+        if isinstance(timestep, torch.Tensor):
+            timestep = timestep.to(self.timesteps.device)
+
+        index_candidates = (self.timesteps == timestep).nonzero()
+
+        # The sigma index that is taken for the **very** first `step`
+        # is always the second index (or the last index if there is only 1)
+        # This way we can ensure we don't accidentally skip a sigma in
+        # case we start in the middle of the denoising schedule (e.g. for image-to-image)
+        if len(index_candidates) > 1:
+            step_index = index_candidates[1]
+        else:
+            step_index = index_candidates[0]
+
+        return step_index.item()
```

Comment:
@patrickvonplaten 
I made the `_init_step_index` to return the index instead of updating the `self._step_index` so that we can use it in the `add_noise` method too. Let me know if it's ok

Reply:
yes that works!

Reply:
Actually I don't think we need to return it here. We don't need to update `add_noise` IMO